controlplane/kubeadm/internal/control_plane.go

```diff
@@ -124,6 +124,17 @@ func (c *ControlPlane) MachineInFailureDomainWithMostMachines(machines Filterabl
 	return machineToMark, nil
 }
 
+// MachineWithDeleteAnnotation returns the machine that has delete annotatation key on it.
+func (c *ControlPlane) MachineWithDeleteAnnotation(machines FilterableMachineCollection) (*clusterv1.Machine, error) {
+	annotatedmachines := machines.Filter(machinefilters.HasAnnotationKey(controlplanev1.DeleteControlPlaneMachineAnnotation))
+	machineToPick := annotatedmachines.Oldest()
+	if machineToPick == nil {
+		return nil, errors.New("failed to pick control plane Machine with delete annotation key")
+	}
+
+	return machineToPick, nil
```

Comment:
Just general question: What will happen if two or more machines have an annotation? We wil return machine with oldest annotation timestamp here?